python/tests/spark/test_functions.py

```diff
@@ -135,51 +135,50 @@ def test_numpy_to_image(spark: SparkSession, tmp_path: Path):
     assert (tmp_path / "1.png").exists()
 
 
-def test_video_to_images(spark: SparkSession):
+def test_video_to_images(
+    spark: SparkSession, tmp_path: Path, asset_path: Path
+):
     """Test extract video frames from YouTubeVideo/VideoStream types
     into list of Image assets.
     """
     sample_rate = 2
     max_samples = 10
-    videostream_df = spark.createDataFrame(
-        [
-            (
-                VideoStream(
-                    uri=os.path.abspath(
-                        os.path.join(
-                            os.path.dirname(__file__),
-                            "..",
-                            "assets",
-                            "big_buck_bunny_short.mp4",
-                        )
-                    )
-                ),
-                Segment(0, 20),
-            ),
-        ],
-        ["video", "segment"],
+    video = VideoStream(str(asset_path / "big_buck_bunny_short.mp4"))
+    df1 = spark.createDataFrame(
+        [(video, Segment(0, 20))], ["video", "segment"]
     )
-    youtube_df = spark.createDataFrame(
-        [
-            (YouTubeVideo(vid="rUWxSEwctFU"), Segment(0, 20)),
-        ],
-        ["video", "segment"],
-    )
-    videostream_df = videostream_df.withColumn(
+    output_dir = tmp_path / "videostream_test"
+    output_dir.mkdir(parents=True)
+    df1 = df1.withColumn(
         "images",
         video_to_images(
-            col("video"), col("segment"), lit(sample_rate), lit(max_samples)
+            col("video"),
+            lit(str(output_dir)),
+            col("segment"),
+            lit(sample_rate),
```

Comment:
I am not expect on this. Can pandas UDF parameter actually be optional ?

Reply:
this isn't a pandas udf. Default arguments work fine i believe (see spectrogram test case)?